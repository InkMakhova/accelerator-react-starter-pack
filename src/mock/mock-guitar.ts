export const mockGuitar = [
  {
    description: 'Практичная, удобная и простая гитара, созданная для музыкантов, только начинающих сфой путь в мир музыки. Она обладает всеми характеристиками, которые сделали работы компании Честер популярными.',
    id: 8,
    name: 'Честер Plus',
    previewImg: 'img/guitar-7.jpg',
    price: 30000,
    rating: 4.5,
    stringCount: 4,
    type: 'electric',
    vendorCode: 'SO135646',
  },
  {
    id: 9,
    name: 'Виолана 300',
    vendorCode: 'VO154751',
    type: 'acoustic',
    description: 'Классическая бюджетная акустическая семиструная гитара от отечественного производителя. Ель в качестве основного материала корпуса отличается лёгкостью и износостойкостью.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 3,
    price: 1700,
  },
  {
    id: 22,
    name: 'Roman TX',
    vendorCode: 'RO214235',
    type: 'ukulele',
    description: 'Бюджетная укулеле. Идеальная для тех, кто только начал своё обучение.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 4,
    rating: 4.5,
    price: 1900,
  },
  {
    id: 7,
    name: 'Dania VX',
    vendorCode: 'DI082347',
    type: 'ukulele',
    description: 'Укулеле - прекрасный компактный аналог классической акустической гитары. Прекрасное качество продуктов компании Dania сочетается с демократичными ценами продуктов линейки GSR.',
    previewImg: 'img/guitar-5.jpg',
    stringCount: 4,
    rating: 2.5,
    price: 2200,
  },
];
