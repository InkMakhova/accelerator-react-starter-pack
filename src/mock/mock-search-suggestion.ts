export const mockSearchSuggestions = [
  {
    id:2,
    name:'Честер Plus',
    vendorCode:'TK129049',
    type:'electric',
    description:'Эргономичность гитары',
    previewImg:'img/guitar-8.jpg',
    stringCount:7,
    rating:3.5,
    price:29500,
  },
  {
    id:1,
    name:'Честер Bass',
    vendorCode:'TK129049',
    type:'electric',
    description:'Эргономичность гитары',
    previewImg:'img/guitar-8.jpg',
    stringCount:7,
    rating:3.5,
    price:29500,
  },
  {
    id:3,
    name:'Честер 6V',
    vendorCode:'TK129049',
    type:'electric',
    description:'Эргономичность гитары',
    previewImg:'img/guitar-8.jpg',
    stringCount:7,
    rating:3.5,
    price:29500,
  },
];
