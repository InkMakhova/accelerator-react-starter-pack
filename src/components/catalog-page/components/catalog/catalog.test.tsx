import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { createMemoryHistory } from 'history';
import Catalog from './catalog';
import { mockGuitar } from '../../../../mock/mock-guitar';

const mockStore = configureMockStore();
const history = createMemoryHistory();

describe('Component: Catalog', () => {
  it('should render Catalog correctly', () => {
    const {container} = render(
      <Provider store={mockStore({
        DATA: {minPrice: '1000', maxPrice: '30000'},
      })}
      >
        <Router history={history}>
          <Catalog guitars={mockGuitar}/>
        </Router>
      </Provider>);
    expect(container).toMatchSnapshot();
  });
});
