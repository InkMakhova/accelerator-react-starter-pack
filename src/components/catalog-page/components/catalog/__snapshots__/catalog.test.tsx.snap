// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Catalog should render Catalog correctly 1`] = `
<div>
  <div
    class="catalog"
  >
    <form
      class="catalog-filter"
    >
      <h2
        class="title title--bigger catalog-filter__title"
      >
        Фильтр
      </h2>
      <fieldset
        class="catalog-filter__block"
      >
        <legend
          class="catalog-filter__block-title"
        >
          Цена, ₽
        </legend>
        <div
          class="catalog-filter__price-range"
        >
          <div
            class="form-input"
          >
            <label
              class="visually-hidden"
            >
              Минимальная цена
            </label>
            <input
              data-testid="min-price"
              id="priceMin"
              name="от"
              placeholder="undefined"
              type="number"
              value=""
            />
          </div>
          <div
            class="form-input"
          >
            <label
              class="visually-hidden"
            >
              Максимальная цена
            </label>
            <input
              data-testid="max-price"
              id="priceMax"
              name="до"
              placeholder="undefined"
              type="number"
              value=""
            />
          </div>
        </div>
      </fieldset>
      <fieldset
        class="catalog-filter__block"
      >
        <legend
          class="catalog-filter__block-title"
        >
          Тип гитар
        </legend>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="acoustic"
            id="acoustic"
            name="acoustic"
            type="checkbox"
          />
          <label
            for="acoustic"
          >
            Акустические гитары
          </label>
        </div>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="electric"
            id="electric"
            name="electric"
            type="checkbox"
          />
          <label
            for="electric"
          >
            Электрогитары
          </label>
        </div>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="ukulele"
            id="ukulele"
            name="ukulele"
            type="checkbox"
          />
          <label
            for="ukulele"
          >
            Укулеле
          </label>
        </div>
      </fieldset>
      <fieldset
        class="catalog-filter__block"
      >
        <legend
          class="catalog-filter__block-title"
        >
          Количество струн
        </legend>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="four"
            id="4-strings"
            name="4-strings"
            type="checkbox"
          />
          <label
            for="4-strings"
          >
            4
          </label>
        </div>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="six"
            id="6-strings"
            name="6-strings"
            type="checkbox"
          />
          <label
            for="6-strings"
          >
            6
          </label>
        </div>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="seven"
            id="7-strings"
            name="7-strings"
            type="checkbox"
          />
          <label
            for="7-strings"
          >
            7
          </label>
        </div>
        <div
          class="form-checkbox catalog-filter__block-item"
        >
          <input
            class="visually-hidden"
            data-testid="twelve"
            id="12-strings"
            name="12-strings"
            type="checkbox"
          />
          <label
            for="12-strings"
          >
            12
          </label>
        </div>
      </fieldset>
    </form>
    <div
      class="catalog-sort"
    >
      <h2
        class="catalog-sort__title"
      >
        Сортировать:
      </h2>
      <div
        class="catalog-sort__type"
      >
        <button
          aria-label="по цене"
          class="catalog-sort__type-button
            catalog-sort__type-button--active"
          data-testid="by-price"
          tabindex="-1"
        >
          по цене
        </button>
        <button
          aria-label="по популярности"
          class="catalog-sort__type-button
            "
          data-testid="by-rating"
          tabindex="0"
        >
          по популярности
        </button>
      </div>
      <div
        class="catalog-sort__order"
      >
        <button
          aria-label="По возрастанию"
          class="catalog-sort__order-button catalog-sort__order-button--up
            catalog-sort__order-button--active"
          data-testid="asc"
          tabindex="-1"
        />
        <button
          aria-label="По убыванию"
          class="catalog-sort__order-button catalog-sort__order-button--down
            "
          data-testid="desc"
          tabindex="0"
        />
      </div>
    </div>
    <div
      class="cards catalog__cards"
    >
      <div
        class="product-card"
      >
        <img
          alt="Честер Plus"
          height="190"
          src="/img/guitar-7.jpg"
          width="75"
        />
        <div
          class="product-card__info"
        >
          <div
            aria-hidden="true"
            class="rate product-card__rate"
          >
            <span
              class="visually-hidden"
            >
              Рейтинг:
            </span>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <span
              class="rate__count"
            >
              4.5
            </span>
            <span
              class="rate__message"
            />
          </div>
          <p
            class="product-card__title"
          >
            Честер Plus
          </p>
          <p
            class="product-card__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            30000
             ₽
          </p>
        </div>
        <div
          class="product-card__buttons"
        >
          <a
            class="button button--mini"
            data-testid="more"
            href="/guitars/8"
          >
            Подробнее
          </a>
          <a
            class="button button--red button--mini button--add-to-cart"
            href="/cart"
          >
            Купить
          </a>
        </div>
      </div>
      <div
        class="product-card"
      >
        <img
          alt="Виолана 300"
          height="190"
          src="/img/guitar-1.jpg"
          width="75"
        />
        <div
          class="product-card__info"
        >
          <div
            aria-hidden="true"
            class="rate product-card__rate"
          >
            <span
              class="visually-hidden"
            >
              Рейтинг:
            </span>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <span
              class="rate__count"
            >
              3
            </span>
            <span
              class="rate__message"
            />
          </div>
          <p
            class="product-card__title"
          >
            Виолана 300
          </p>
          <p
            class="product-card__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            1700
             ₽
          </p>
        </div>
        <div
          class="product-card__buttons"
        >
          <a
            class="button button--mini"
            data-testid="more"
            href="/guitars/9"
          >
            Подробнее
          </a>
          <a
            class="button button--red button--mini button--add-to-cart"
            href="/cart"
          >
            Купить
          </a>
        </div>
      </div>
      <div
        class="product-card"
      >
        <img
          alt="Roman TX"
          height="190"
          src="/img/guitar-8.jpg"
          width="75"
        />
        <div
          class="product-card__info"
        >
          <div
            aria-hidden="true"
            class="rate product-card__rate"
          >
            <span
              class="visually-hidden"
            >
              Рейтинг:
            </span>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <span
              class="rate__count"
            >
              4.5
            </span>
            <span
              class="rate__message"
            />
          </div>
          <p
            class="product-card__title"
          >
            Roman TX
          </p>
          <p
            class="product-card__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            1900
             ₽
          </p>
        </div>
        <div
          class="product-card__buttons"
        >
          <a
            class="button button--mini"
            data-testid="more"
            href="/guitars/22"
          >
            Подробнее
          </a>
          <a
            class="button button--red button--mini button--add-to-cart"
            href="/cart"
          >
            Купить
          </a>
        </div>
      </div>
      <div
        class="product-card"
      >
        <img
          alt="Dania VX"
          height="190"
          src="/img/guitar-5.jpg"
          width="75"
        />
        <div
          class="product-card__info"
        >
          <div
            aria-hidden="true"
            class="rate product-card__rate"
          >
            <span
              class="visually-hidden"
            >
              Рейтинг:
            </span>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="11"
              width="12"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <span
              class="rate__count"
            >
              2.5
            </span>
            <span
              class="rate__message"
            />
          </div>
          <p
            class="product-card__title"
          >
            Dania VX
          </p>
          <p
            class="product-card__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            2200
             ₽
          </p>
        </div>
        <div
          class="product-card__buttons"
        >
          <a
            class="button button--mini"
            data-testid="more"
            href="/guitars/7"
          >
            Подробнее
          </a>
          <a
            class="button button--red button--mini button--add-to-cart"
            href="/cart"
          >
            Купить
          </a>
        </div>
      </div>
    </div>
    <div
      class="pagination page-content__pagination"
    >
      <ul
        class="pagination__list"
      >
        
        <li
          class="pagination__page pagination__page--active"
        >
          <a
            class="link pagination__page-link"
            data-testid="1"
            href="/catalog/page_1"
          >
            1
          </a>
        </li>
        <li
          class="pagination__page "
        >
          <a
            class="link pagination__page-link"
            data-testid="2"
            href="/catalog/page_2"
          >
            2
          </a>
        </li>
        <li
          class="pagination__page "
        >
          <a
            class="link pagination__page-link"
            data-testid="3"
            href="/catalog/page_3"
          >
            3
          </a>
        </li>
        
      </ul>
    </div>
  </div>
</div>
`;
